dask_pytorch/results.py

```diff
@@ -0,0 +1,91 @@
+"""
+Infrastructure for retrieving and logging intermediate results from pytorch training jobs.
+
+Currently using dask pub/sub, but will create an S3 version in the future.
+"""
+import uuid
+import logging
+import os
+from typing import List
+from os.path import join, exists, dirname
+
+from distributed.pubsub import Pub, Sub
+from distributed.utils import TimeoutError  # pylint: disable=redefined-builtin
+from distributed.client import wait, FIRST_COMPLETED, Future
+
+
+logger = logging.getLogger(__name__)
+
+
+class DaskResultsHandler:
+    """
+    This class use Dask pubsub infra to pass intermediate results back from PyTorch
+    jobs to the client.
+    """
+
+    def __init__(self, pub_sub_key: str = None):
+        """
+        pub_sub_key is an arbitrary string (topic) for the pub sub channel.
+        It's a good idea to change it.  Sometimes old topics can get "clogged"
+        """
+        if pub_sub_key is None:
+            pub_sub_key = uuid.uuid4().hex
+        self.pub_sub_key = pub_sub_key
+
+    @classmethod
+    def _get_all(cls, sub: Sub):
+        while True:
+            try:
+                yield sub.get(timeout=1.0)
+            except TimeoutError:
+                break
+
+    def _get_results(self, futures: List[Future], raise_errors: bool = True):
+        sub = Sub(self.pub_sub_key)
+        while True:
+            for obj in self._get_all(sub):
+                yield obj
+            if not futures:
+                break
+            try:
+                result = wait(futures, 0.1, FIRST_COMPLETED)
+            except TimeoutError:
+                continue
+
+            for fut in result.done:
+                try:
+                    fut.result()
+                except Exception as e:  # pylint: disable=broad-except
+                    if raise_errors:
+                        raise
+                    logging.exception(e)
```

Comment:
```suggestion
                    logging.exception(e)
                    if raise_errors:
                        raise
```

I think you should do the logging part first, otherwise this exception won't make it into the logs if `raise_errors = True`.